front/src/modules/trainschedule/components/TimetableV2/hooks.ts

```diff
@@ -80,6 +82,21 @@ const useTrainSchedulesDetails = (
           if (validityFilter === 'invalid' && trainSummary.status === 'success') return false;
         }
 
+        // Apply scheduled points honored filter
+        if (scheduledPointsHonoredFilter !== 'both') {
+          if (trainSummary.status === 'success') {
+            const isHonored = trainSummary.scheduled_points_honored;
+            if (
+              (scheduledPointsHonoredFilter === 'honored' && !isHonored) ||
+              (scheduledPointsHonoredFilter === 'notHonored' && isHonored)
+            ) {
+              return false;
+            }
+          } else if (scheduledPointsHonoredFilter === 'honored') {
```

Comment:
I think that we should filter all the trains that are invalid when we use the filter scheduledPointsHonored. I check this with Giuliana
```suggestion
          } else {
```

Reply:
Done